helios-services/src/test/java/com/spotify/helios/agent/TaskRunnerTest.java

```diff
@@ -113,4 +129,49 @@ public void testPullTimeoutVariation() throws Throwable {
       assertEquals(ImagePullFailedException.class, t.getCause().getClass());
     }
   }
+
+  @Test
+  public void testContainerNotRunningVariation() throws Throwable {
+
+    final ContainerInfo stopped = new ContainerInfo() {
+      @Override
+      public ContainerState state() {
+        final ContainerState state = mock(ContainerState.class);
+        when(state.running()).thenReturn(false);
+        when(state.error()).thenReturn("container is a potato");
+        return state;
+      }
+    };
+
+    doReturn("potato").when(mockCreation).id();
```

Comment:
any particular reason to use this style instead of `when(mockCreation.id()).thenReturn("potato")` ?